SecondHomework/Logistic/ParcelManager.cs

```diff
@@ -13,38 +21,39 @@ public enum WeightCategory
         MoreThan20Kg
     };
 
-    public ParcelManager()
+    public ParcelManager(FileManager fileManager)
     {
+        _fileManager = fileManager;
         _parcels = new List<Parcel>();
     }
 
-    public ParcelManager(FileManager fileManager)
+    public async Task InitializeAsync()
     {
-        _fileManager = fileManager;
-        _parcels = fileManager.Read();
+        _parcels = await _fileManager.ReadAsync();
     }
 
-    public void AddParcel(Parcel parcel)
+    public async Task AddParcelAsync(Parcel parcel)
     {
         _parcels.Add(parcel);
-        _fileManager.Save(_parcels);
+        await _fileManager.SaveParcelsAsync(_parcels);
     }
 
-    public List<Parcel> GetParcels()
+    public async Task<List<Parcel>> GetParcelsAsync()
     {
+        _parcels = await _fileManager.ReadAsync();
```

Comment:
Why you need to read it on every read? Just read it from file in constructor because you update collection in memory during all other operations